consensus/istanbul/backend/api.go

```diff
@@ -42,6 +42,21 @@ type API struct {
 }
 
 // getHeaderByNumber retrieves the header requested block or current if unspecified.
+func (api *API) getHeaderByNumber(number *rpc.BlockNumber) (*types.Header, error) {
+	if number == nil || *number == rpc.LatestBlockNumber || *number == rpc.PendingBlockNumber {
+		head := api.chain.CurrentHeader()
+		if head == nil {
+			return nil, errUnknownBlock
+		}
+		return head, nil
+	} else if *number == rpc.EarliestBlockNumber {
+		return api.chain.GetHeaderByNumber(0), errUnknownBlock
```

Comment:
This `errUnknownblock` doesn't sound right.  I guess it should also check against nil and return error if appropriate